torch/csrc/cudnn/Conv.cpp

```diff
@@ -482,18 +483,18 @@ Workspace chooseAlgorithm(
   }
 }
 
+
 void* tensorPointer(
-    cudnnDataType_t dataType, THVoidTensor* tensor, int groupIdx, int groups,
+    cudnnDataType_t dataType, const at::Tensor& tensor, int groupIdx, int groups,
     int dim)
 {
   int elementSize = dataSize(dataType);
-  char* ptr = (char*) tensor->storage->data;
-  ptr += elementSize * tensor->storageOffset;
+  char* ptr = (char*) tensor.data_ptr();
 #if CUDNN_VERSION < 7000
   if (groupIdx > 0) {
     size_t size = 1;
-    for (int i = dim; i < tensor->nDimension; ++i) {
-      size *= tensor->size[i];
+    for (int i = dim; i < tensor.dim(); ++i) {
+      size *= tensor.size(i);
```

Comment:
I think that this function could be simplified. The problem before was that we had no indexing functions for `THVoidTensor`, but `at::Tensor`s should have them implemented. This function is basically a narrow along `dim`

Reply:
Let me handle this when I get back from China, and don't have to go over a slow ssh connection to a GPU machine to test these changes :) (typechecker only development, aww yeahhhh)